File: monitoring/src/main/java/com/example/monitoring/user/application/service/UserService.java
package com.example.monitoring.user.application.service;

import com.example.monitoring.user.application.port.in.UserUseCase;
import com.example.monitoring.user.application.port.in.request.CreateUserRequestDto;
import com.example.monitoring.user.application.port.in.response.UserResponseDto;
import com.example.monitoring.user.application.port.out.UserRepository;
import com.example.monitoring.user.domain.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
class UserService implements UserUseCase {

    private final UserRepository userRepository;

    @Override
    public UserResponseDto findById(Long userId) {
        User user = userRepository.findById(userId)
                .orElseThrow();
        return UserResponseDto.from(user);
    }

    @Override
    public UserResponseDto createUser(CreateUserRequestDto createUserRequestDto) {
        User user = createUserRequestDto.createUser();
        User savedUser = userRepository.save(user);
        return UserResponseDto.from(savedUser);
    }
}

File: kotlin-spring-boot/src/main/kotlin/com/hello/kotlin/user/service/UserService.kt
package com.hello.kotlin.user.service

import com.hello.kotlin.user.domain.UserRepository
import com.hello.kotlin.user.service.dto.request.CreateUserRequestDto
import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Service
@Transactional
class UserService (
    private val userRepository: UserRepository
) : UserUseCase {
    override fun save(createUserRequestDto: CreateUserRequestDto) {

    }
}